packages/shared-ux/chrome/navigation/src/ui/v2/components/navigation.tsx

```diff
@@ -63,44 +89,39 @@ export function Navigation({ children, onRootItemRemove }: Props) {
       });
 
       return {
-        unregister: () => {
-          if (onRootItemRemove) {
-            onRootItemRemove(navNode.id);
-          }
-
-          setNavigationItems((prevItems) => {
-            const updatedItems = { ...prevItems };
-            delete updatedItems[navNode.id];
-            return updatedItems;
-          });
-        },
-        path: [],
+        unregister,
+        path: [navNode.id],
       };
     },
-    [onRootItemRemove]
+    [unregister]
   );
 
   const contextValue = useMemo<Context>(
     () => ({
       register,
+      updateFooterChildren: setFooterChildren,
+      unstyled,
     }),
-    [register]
+    [register, unstyled]
   );
 
   useEffect(() => {
-    // Send the navigation tree to the Chrome service
+    // This will update the navigation tree in the Chrome service (calling the serverless.setNavigation())
     onProjectNavigationChange({
```

Comment:
This seems to be getting called every time register/unregister is called. So currently we notify the outside about navigation changes while we are in the process of rendering the tree. 

This is visible in storybook as `Update chrome navigation` is called 8 times during the initial render: 

![Screenshot 2023-06-02 at 11 33 22](https://github.com/elastic/kibana/assets/7784120/6604954d-6bf4-4dbd-b36f-2cccfefe1196)


I think this should be fixed / batched somehow

Reply:
Before optimizing too early we'd need to know what we are trying to optimize for 😊 
This is similar to how a react component might render 8 times because multiple inside states have changed. No big deal.

At this point I don't think this is a problem. We can revisit if we see a perf issue somewhere.

Reply:
I think this is a bug, not just performance optimization. The component notifies about navigation tree changes while it is gathering the navigation tree. It calls `onProjectNavigationChange` with an intermediate tree.

Reply:
That's how react works :) I would not consider this a "bug" until it the business rule of the application does not work. If we were doing 8 HTTP requests then yeah we should avoid that. At this stage I am not seeing yet what are the implication. Like I said, we can revisit in other PR if there is a need to optimise and batch.